opencensus_shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetryCtx.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.opencensusshim;
+
+import io.opencensus.trace.ContextHandle;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+
+public class OpenTelemetryCtx implements ContextHandle {
+
+  private final Context context;
+  private Scope scope;
+
+  public OpenTelemetryCtx(Context context) {
+    this.context = context;
+  }
+
+  Context getContext() {
+    return context;
+  }
+
+  @Override
+  public ContextHandle attach() {
+    scope = context.makeCurrent();
+    return this;
+  }
+
+  @Override
+  public void detach(ContextHandle ctx) {
+    OpenTelemetryCtx impl = (OpenTelemetryCtx) ctx;
```

Comment:
Note that we are closing the scope of a context passed as an argument here, not the scope of `this.scope`.
Which seems to imply that we should implement a "nesting" mechanism (using a Stack data structure probably), e.g.

```
create context
-- attach - creates root scope 1
   -- attach - creates child scope 2
       -- attach creates a sub-child scope 3
       -- detach closes scope 3 and restores scope 2 as current
   -- detach closes scope 2 and restores scope 1 as current
-- detach closes scope 1 and there is no current scope anymore
```

Reply:
The OpenTelemetry context has the same semantics as the grpc context (the implementation was heavily inspired by it), so I think this should all "just work" as long as everything is kept in alignment the same way.